projects/core/src/occ/adapters/user/occ-user-interests.adapter.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { OccEndpointsService } from '../../services/occ-endpoints.service';
+import { Observable, throwError, forkJoin } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { UserInterestsAdapter } from '../../../user/connectors/interests/user-interests.adapter';
+import {
+  ProductInterestSearchResult,
+  ProductInterestEntryRelation,
+  NotificationType,
+} from '../../../model/product-interest.model';
+import { OccConfig } from '../../config/occ-config';
+import { ConverterService } from '../../../util/converter.service';
+import { PRODUCT_INTERESTS_NORMALIZER } from '../../../user/connectors/interests/converters';
+
+@Injectable()
+export class OccUserInterestsAdapter implements UserInterestsAdapter {
+  constructor(
+    protected http: HttpClient,
+    protected occEndpoints: OccEndpointsService,
+    protected config: OccConfig,
+    protected converter: ConverterService
+  ) {}
+
+  public getInterests(
+    userId: string,
+    pageSize?: number,
+    currentPage?: number,
+    sort?: string,
+    productCode?: string,
+    notificationType?: NotificationType
+  ): Observable<ProductInterestSearchResult> {
+    let params = new HttpParams()
+      .set('sort', sort ? sort : 'name:asc')
+      .set('fields', 'FULL');
```

Comment:
We should consider including `fields` definition in endpoint configuration, so customer would be able to customize it.  And because we use the same endpoint for other actions (like delete and update) that don't require `fields` we need to create separate endpoint for get method, could be named: `getProductInterests`.

Reply:
Fixed.